### ISSUE TITLE
Named tasks (actions/operations)

### ISSUE THREAD

**Issue by ggeorgiev** [concourse/concourse#1016]
# Feature Request

## What challenge are you facing?

I am using https://github.com/cloudfoundry-community/slack-notification-resource for a while and I had several inconveniences to deal with.

1. Even it has check no-op there is a container that is always active, so I had to also use check_every to eliminate that.
2. Every time I use the side effect of the resource I have two extra containers for `check` and `get`.
3. It is shown as output in the pipeline that I would like to hide.

Recently I start using https://github.com/troykinsella/concourse-fly-resource and yet exactly the same issues. I asked myself, "why are those resources?" after all they do an action for which they predefine an image - isn't this exactly what a task is for?

## A Modest Proposal

One option is obviously to make the resource concept more flexible to accumulate for such type of "resources". 

Another is concourse to allow for such glorified tasks. Sometimes perception is everything. No one would like to put an effort for creating an image and a script for concourse use. But creating concourse action/operation/named-task or whatever name for such concept is given can make a lot more sense - even at the end being the same.

The long story short - I am suggesting a new way to define tasks to be introduced that follows the resource model. A named shared configuration that provides the `image` and `target` and then an `execute` step in the job plans that could add additional arguments.

This might sound as a syntactic sugar, but the sweetness in life should not be underestimated.

**Comment by joshzarrabi** [comment]
@ggeorgiev I would argue that resources are already very flexibly defined, as in your first idea in your proposal. They are simply things that do other things. Even some of our base resource types, like the time resource, don't actually store or get any information, but they do follow the resource interface. This provides a simple abstraction, and makes it easy to write new resources that do what you want.

**Comment by ggeorgiev** [comment]
what about the three issues with using resources:

1. Even it has check no-op there is a container that is always active, so I had to also use check_every to eliminate that.
2. Every time I use the side effect of the resource I have two extra containers for check and get.
3. It is shown as output in the pipeline that I would like to hide.

**Comment by joshzarrabi** [comment]
Those are fair issues, but I really do not see us changing this for the time being, as the cost would be really high. It would be a large implementation cost, and we would have to re-educate users on the best practice between using the special tasks and resources. I definitely see where you are coming from, and I have had similar thoughts before, but as a member of the engineering team I do not think it would be worth the effort for us or our users.
